plugins/flatpak/gs-flatpak.c

```diff
@@ -1380,6 +1445,7 @@ gs_flatpak_add_updates_pending (GsFlatpak *self, GsAppList *list,
 		guint64 download_size = 0;
 		g_autoptr(GsApp) app = NULL;
 		g_autoptr(GError) error_local = NULL;
+		GsApp *main_app = NULL;
```

Comment:
You're leaking `main_app` in this function too

Reply:
Still applies